frontend/src/app/sessions/active-sessions/upload-dialog/upload-dialog.component.ts

```diff
@@ -249,6 +251,19 @@ export class UploadDialogComponent implements OnInit, OnDestroy {
       });
   }
 
+  download(filename: string) {
+    this.session.download_in_progress = true;
+    this.loadService
+      .download(this.session.id, filename)
+      .subscribe((response: Blob) => {
+        saveAs(
+          response,
+          `${filename.replace(/^[\/\\: ]+/, '').replace(/[\/\\: ]+/g, '_')}.zip`
+        );
+        this.session.download_in_progress = false;
+      });
```

Comment:
The subscription should be destroyed on `ngOnDestroy` (someone starts an download, but closes the dialog).

Reply:
Done. Not sure if it will have negative side effects when a big model is downloaded and the dialog is closed.